### PR TITLE
refactor: Removes the deprecated DASHBOARD_NATIVE_FILTERS feature flag

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
As part of the [4.0 approved initiatives](https://github.com/orgs/apache/projects/292/views/1), this PR removes the `DASHBOARD_NATIVE_FILTERS` feature flag and enables the feature permanently.

The previous value of the feature flag was `True` and now the feature is permanently enabled.

### TESTING INSTRUCTIONS
CI should be sufficient for merging this PR. We'll do a complete testing of 4.0 after all approved proposals are merged.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/plugins/plugin-chart-handlebars/src/plugin/controls/pagination.tsx [hunk 1]
```diff
@@ -24,23 +24,23 @@ import {
 import { FeatureFlag, isFeatureEnabled, t } from '@superset-ui/core';
 import { PAGE_SIZE_OPTIONS } from '../../consts';
 
-export const serverPaginationControlSetRow: ControlSetRow =
```
Comment (geido) [inline superset-frontend/plugins/plugin-chart-handlebars/src/plugin/controls/pagination.tsx:27]: @michael-s-molina I am not too sure about this change. It seems that server_pagination should be shown by default now as the  `DASHBOARD_NATIVE_FILTERS` is also by default

File: superset-frontend/src/dataMask/actions.ts [hunk 2]
```diff
@@ -74,9 +74,9 @@ export function updateDataMask(
   dataMask: DataMask,
 ): UpdateDataMask {
   // Only apply data mask if one of the relevant features is enabled
-  const isFeatureFlagActive =
-    isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS) ||
-    isFeatureEnabled(FeatureFlag.DASHBOARD_CROSS_FILTERS);
+  const isFeatureFlagActive = isFeatureEnabled(
```
Comment (geido) [inline superset-frontend/src/dataMask/actions.ts:77]: Same comment as above

File: UPDATING.md [hunk 3]
```diff
@@ -29,6 +29,8 @@ assists people when migrating to a new version.
 
 ### Breaking Changes
 
+- [X](https://github.com/apache/superset/issues/X): Removes the deprecated `DASHBOARD_NATIVE_FILTERS` feature flag. The previous value of the feature flag was `True` and now the feature is permanently enabled.
```
Comment (john-bodley) [inline UPDATING.md:32]:
```suggestion
- [26329](https://github.com/apache/superset/pull/26329): Removes the deprecated `DASHBOARD_NATIVE_FILTERS` feature flag. The previous value of the feature flag was `True` and now the feature is permanently enabled.
```

File: superset-frontend/src/dashboard/actions/sliceEntities.ts [hunk 4]
```diff
@@ -114,13 +108,11 @@ export function fetchSlices(
...
       : [];
 
-    if (isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS)) {
-      filters.push({
-        col: 'viz_type',
-        opr: 'neq',
-        value: 'filter_box',
-      });
-    }
+    filters.push({
```
Comment (john-bodley) [inline superset-frontend/src/dashboard/actions/sliceEntities.ts:111]: Do we still need this logic given that filter boxes are slatted to be removed in https://github.com/apache/superset/pull/26328?

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 5]
```diff
@@ -383,10 +381,7 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
             />
           </FormItem>
         )}
-        {!(
-          isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS) &&
-          this.state.vizType === 'filter_box'
-        ) && (
+        {!(this.state.vizType === 'filter_box') && (
```
Comment (john-bodley) [inline superset-frontend/src/explore/components/SaveModal.tsx:384]: See previous comment.

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 6]
```diff
@@ -456,8 +451,7 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
           !this.state.dashboard ||
           (this.props.datasource?.type !== DatasourceType.Table &&
             !this.state.datasetName) ||
-          (isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS) &&
-            this.state.vizType === 'filter_box')
+          this.state.vizType === 'filter_box'
```
Comment (john-bodley) [inline superset-frontend/src/explore/components/SaveModal.tsx:454]: See previous comment.

File: superset-frontend/src/explore/components/controls/VizTypeControl/index.tsx [hunk 7]
```diff
@@ -48,10 +46,7 @@ const bootstrapData = getBootstrapData();
 const denyList: string[] = bootstrapData.common.conf.VIZ_TYPE_DENYLIST || [];
 const metadataRegistry = getChartMetadataRegistry();
 
-if (
-  isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS) &&
-  !denyList.includes('filter_box')
-) {
+if (!denyList.includes('filter_box')) {
```
Comment (john-bodley) [inline superset-frontend/src/explore/components/controls/VizTypeControl/index.tsx:49]: See previous comment.

File: superset-frontend/src/pages/ChartCreation/index.tsx [hunk 8]
```diff
@@ -64,10 +62,7 @@ const ELEMENTS_EXCEPT_VIZ_GALLERY = ESTIMATED_NAV_HEIGHT + 250;
 const bootstrapData = getBootstrapData();
 const denyList: string[] = bootstrapData.common.conf.VIZ_TYPE_DENYLIST || [];
 
-if (
-  isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS) &&
-  !denyList.includes('filter_box')
-) {
+if (!denyList.includes('filter_box')) {
```
Comment (john-bodley) [inline superset-frontend/src/pages/ChartCreation/index.tsx:65]: See previous comment.

File: superset-frontend/plugins/plugin-chart-table/src/controlPanel.tsx [hunk 9]
```diff
@@ -290,8 +290,7 @@ const config: ControlPanelConfig = {
             },
           },
         ],
-        isFeatureEnabled(FeatureFlag.DASHBOARD_CROSS_FILTERS) ||
-        isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS)
+        isFeatureEnabled(FeatureFlag.DASHBOARD_CROSS_FILTERS)
```
Comment (justinpark) [inline superset-frontend/plugins/plugin-chart-table/src/controlPanel.tsx:293]: This condition ought to be disregarded as it is constantly satisfied due to its association with an 'OR' condition before.

File: superset-frontend/src/dashboard/actions/sliceEntities.ts (hunk 4 above)
Comment (michael-s-molina) [reply superset-frontend/src/dashboard/actions/sliceEntities.ts:111]: Although related, the features work independently which allows us to remove them in separate PRs. Anything related to filter boxes will be removed in a specific PR.

File: superset-frontend/plugins/plugin-chart-table/src/controlPanel.tsx (hunk 9 above)
Comment (michael-s-molina) [reply superset-frontend/plugins/plugin-chart-table/src/controlPanel.tsx:293]: Makes sense. Done.

**Review by kgabryje — APPROVED** [review]
LGTM

**Review by justinpark — APPROVED** [review]
LGTM

**Pull request merged**